pulsar/client.go

```diff
@@ -91,6 +92,9 @@ type ClientOptions struct {
 
 	// Max number of connections to a single broker that will kept in the pool. (Default: 1 connection)
 	MaxConnectionsPerBroker int
+
+	// Configure the logger used by the client. (Default: logrus.StandardLogger())
+	Logger *logrus.Logger
```

Comment:
I like this approach, though in this case we'd be exposing Logrus as part of our API, something that we cannot later rollback. 

If we want to take a logger option in, we should have a custom logger interface (with default impl bound to logrus).

Reply:
Yeah, i agree with you. It may need some effort to extract a reasonable logger interface, how about resolving it in another PR?

Reply:
My main concern is that once we expose in API, people can depend on it. So we should rather first define the logger API and then apply this change.

Reply:
Agreed, will add logger and entry interface later.

Reply:
it makes sense, i defined the `Logger` and `Entry` interfaces at `pulsar/log` package, please take a look :-)